app/helpers/heatmap_helper.rb

```diff
@@ -1,61 +1,25 @@
+# See selectedOptions in SamplesHeatmapView for client-side defaults, and
+# heatmap action in VisualizationsController.
 module HeatmapHelper
   DEFAULT_MAX_NUM_TAXONS = 30
+  # Zscore is best for heatmaps because it weighs the frequency against the background
+  DEFAULT_TAXON_SORT_PARAM = 'highest_nt_zscore'.freeze
+  READ_SPECIFICITY = true
+  MINIMUM_READ_THRESHOLD = 5
+  # Note: this is activated from the heatmap page by selecting "Viruses -
+  # Phages". The default categories are all BUT phages, though the UI does not
+  # indicate this.
+  INCLUDE_PHAGE = false
 
-  def heatmap
-    {
-      taxonLevels: %w[Genus Species],
-      categories: ReportHelper::ALL_CATEGORIES.pluck('name'),
-      subcategories: {
-        Viruses: ["Phage"]
-      },
-      metrics: [
-        { text: "Aggregate Score", value: "NT.aggregatescore" },
-        { text: "NT Z Score", value: "NT.zscore" },
-        { text: "NT rPM", value: "NT.rpm" },
-        { text: "NT r (total reads)", value: "NT.r" },
-        { text: "NR Z Score", value: "NR.zscore" },
-        { text: "NR r (total reads)", value: "NR.r" },
-        { text: "NR rPM", value: "NR.rpm" }
-      ],
-      backgrounds: current_power.backgrounds.map do |background|
-        { name: background.name, value: background.id }
-      end,
-      thresholdFilters: {
-        targets: [
-          { text: "Aggregate Score", value: "NT_aggregatescore" },
-          { text: "NT Z Score", value: "NT_zscore" },
-          { text: "NT rPM", value: "NT_rpm" },
-          { text: "NT r (total reads)", value: "NT_r" },
-          { text: "NT %id", value: "NT_percentidentity" },
-          { text: "NT L (alignment length in bp)", value: "NT_alignmentlength" },
-          { text: "NT log(1/e)", value: "NT_neglogevalue" },
-          { text: "NR Z Score", value: "NR_zscore" },
-          { text: "NR r (total reads)", value: "NR_r" },
-          { text: "NR rPM", value: "NR_rpm" },
-          { text: "NR %id", value: "NR_percentidentity" },
-          { text: "NR L (alignment length in bp)", value: "NR_alignmentlength" },
-          { text: "R log(1/e)", value: "NR_neglogevalue" }
-        ],
-        operators: [">=", "<="]
-      },
-      allowedFeatures: current_user.allowed_feature_list
-    }
-  end
-
-  def download_heatmap
-    @sample_taxons_dict = sample_taxons_dict(params)
-    output_csv = generate_heatmap_csv(@sample_taxons_dict)
-    send_data output_csv, filename: 'heatmap.csv'
-  end
+  MINIMUM_ZSCORE_THRESHOLD = 1.7
 
-  def samples_taxons
-    @sample_taxons_dict = sample_taxons_dict(params)
-    render json: @sample_taxons_dict
-  end
+  def self.sample_taxons_dict(params, samples)
+    return {} if samples.empty?
 
-  def sample_taxons_dict(params)
-    sample_ids = (params[:sampleIds] || []).map(&:to_i)
     num_results = params[:taxonsPerSample] ? params[:taxonsPerSample].to_i : DEFAULT_MAX_NUM_TAXONS
+    min_reads = params[:minReads] ? params[:minReads].to_i : MINIMUM_READ_THRESHOLD
```

Comment:
this is now dynamic just so I could have fewer fixtures in the unit test

Reply:
One of the reasons we are moving to factories... need to get my PR out soon.